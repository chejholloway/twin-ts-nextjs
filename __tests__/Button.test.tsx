/* eslint-disable @typescript-eslint/no-unused-vars */
import tw, { styled, css, theme } from 'twin.macro';
import { expect, jest, test } from '@jest/globals';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

jest.mock('twin.macro');

const PrimaryButtonTest = `<Button variant="primary">Submit</Button>`;
const SecondaryButtonTest = `<Button variant="secondary">Cancel</Button>`;
const SmallButtonTest = `<Button isSmall>Close</Button>`;

describe('Button Tests', () => {
  it('Primary Button renders correctly', () => {
    expect(renderer.create(PrimaryButtonTest).toJSON()).toMatchSnapshot();
  });

  it('Secondary Button renders correctly', () => {
    expect(renderer.create(SecondaryButtonTest).toJSON()).toMatchSnapshot();
  });

  it('Small Button renders correctly', () => {
    expect(renderer.create(SmallButtonTest).toJSON()).toMatchSnapshot();
  });
});
