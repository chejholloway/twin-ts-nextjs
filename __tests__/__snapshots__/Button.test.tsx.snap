// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests Primary Button renders correctly 1`] = `"<Button variant="primary">Submit</Button>"`;

exports[`Button Tests Secondary Button renders correctly 1`] = `"<Button variant="secondary">Cancel</Button>"`;

exports[`Button Tests Small Button renders correctly 1`] = `"<Button isSmall>Close</Button>"`;
